whylog/assistant/regex_assistant/regex.py

```diff
@@ -3,13 +3,75 @@
 """
 
 import re
+from collections import deque
+
+import six
 
 from whylog.assistant.regex_assistant.exceptions import NotMatchingRegexError
+from whylog.assistant.span import Span
+from whylog.assistant.span_list import SpanList
 
 special_characters = frozenset('.^$*+{}?[]|()]')
 group_pattern = re.compile(r"[a-zA-Z]+|[0-9]+|\s+|[\W]+")
 
 
+def group_spans_from_regex(regex, text):
+    """
+    Creates spans from regex groups in regex corresponding to text.
+
+    Assumption: regex matches text.
+    """
+
+    if len(regex) < 2:
+        return []
+
+    verify_regex(regex, text)
+
+    matcher = re.match(re.compile(regex), text)
```

Comment:
do not compile it here like that. Who calls this function? Can he compile the regex before passing it here?